backend/src/routes/api/notebooks/index.ts

```diff
@@ -61,42 +63,59 @@ module.exports = async (fastify: KubeFastifyInstance) => {
   );
 
   fastify.post(
-    '/:namespace',
+    '/create',
     secureRoute(fastify)(
       async (
         request: FastifyRequest<{
-          Params: {
-            namespace: string;
-          };
-          Body: Notebook;
+          Body: NotebookData;
         }>,
+        reply: FastifyReply,
       ) => {
-        return createNotebook(fastify, request);
+        const notebookData = request.body;
+        const username = await getUserName(fastify, request);
+
+        return createNotebook(fastify, notebookData, username).catch((e) => {
+          fastify.log.error(
+            `Failed to create notebook, ${e.response?.data?.message || e.message}}`,
+          );
+          reply.status(400).send(e.response?.data?.message || e.message);
+        });
       },
     ),
   );
 
-  fastify.patch(
-    '/:namespace/:name',
+  fastify.put(
+    '/restart',
```

Comment:
PATCH {state: `stop`}

Reply:
Starting needs data -- we have a stale Notebook, you need to be able to accept new configurations when we start it back up.

Reply:
good point